external-crates/move/crates/move-vm-runtime/src/loader/ast.rs

```diff
@@ -88,6 +88,619 @@ pub(crate) struct LoadedModule {
     pub(crate) instantiation_signatures: BTreeMap<SignatureIndex, Vec<Type>>,
 }
 
+// A runtime function
+// #[derive(Debug)]
+// https://github.com/rust-lang/rust/issues/70263
+pub(crate) struct Function {
```

Comment:
I am going to leave a comment about this just for you (and us all) to think about, no work or action to take right now.
What follows is more of a intuition/feeling/obnoxiousness on my part, but there...
I was thinking that `Function` as this is defined was not allocated in the arena but in a classic normal rust code/struct/container.
What would be in the arena would be the minimal info to set up the frame.

I understand that creates a problem for debug and core dump functions (maybe) but I liked the idea that Arena contained only "code", assuming that make sense.

Anyway I will explore that myself in a near future to see if that is even possible and not a big deal anyway.

When compiling I would assume we would not hit the arenas to find info (metadata) about a function